dowhy/gcm/influence.py

```diff
@@ -0,0 +1,339 @@
+                             noise terms and the target node. This can be an instance of a PredictionModel, the string
+                             'approx' or the string 'exact'. With 'exact', the underlying causal models in the graph
+                             are utilized directly by propagating given noise inputs through the graph. This is
+                             generally more accurate but slow. With 'approx', an appropriate model is selected and
+                             trained based on sampled data from the graph, which is less accurate but faster. A more
+                             detailed treatment on why we need this parameter is also provided in :ref:`icc`.
+    :param attribution_func: Optional attribution function to measure the statistical property of the target node. This
+                             function expects two inputs; predictions after the randomization of certain features (i.e.
+                             samples from noise nodes) and a baseline where no features were randomized. The baseline
+                             predictions can be typically ignored if one is interested in uncertainty measures such as
+                             entropy or variance, but they might be relevant if, for instance, these shall be estimated
+                             based on the residuals. By default, entropy is used if prediction model is a classifier,
+                             variance otherwise.
+    :param num_training_samples: Number of samples for fitting the prediction_model. This number should be larger
+                                 than num_noise_samples + num_evaluation_samples.
+    :param num_noise_samples: Size of noise samples of upstream noise terms (of the target node) to draw from the
+                              causal graph. These will be used as background samples.
+    :param num_evaluation_samples: Size of noise samples to use in computing intrinsic causal contribution. These
+                                   will be used as samples of interest.
+    :param max_batch_size: Maximum batch size for estimating the predictions from evaluation samples. This has a
+                           significant impact on the overall memory usage. If set to -1, all samples are used in one
+                           batch.
+    :param approx_select_model: Optional function to select the prediction_model when prediction_model is set to
+                                'approx'. If not provided, defaults to auto.select_model().
+    :param shapley_config: Configuration for the Shapley estimator.
+    :return: Intrinsic causal contribution of each ancestor node to the statistical property defined by the
+             attribution_func of the target node.
+    """
+    validate_causal_dag(causal_model.graph)
+
+    reduced_causal_model = StructuralCausalModel(node_connected_subgraph_view(causal_model.graph, target_node))
```

Comment:
good to have an inline comment defining the operation that leads to "reduced" cm

Reply:
Added docstring and renamed it to ```sub_causal_model``` to be consistent.